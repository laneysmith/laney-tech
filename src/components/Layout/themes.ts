export const light = {
  backgroundColor: '#FFFFFF',
  textColor: '#24292e',
  secondaryTextColor: '#50555a',
  linkColor: '#0366d6',
  gridBackgroundColor: '#FFFFFF',
  cellColor: 'rgb(235, 237, 240)',
  borderColor: '#d1d5da',
  accentColor1: '#c6e48b',
  accentColor2: '#7bc96f',
  accentColor3: '#239a3b',
  accentColor4: '#196127',
  accentColor5: '#196127',
};

export const dark = {
  backgroundColor: '#1E1E1E',
  textColor: '#D2D2D2',
  secondaryTextColor: '#D2D2D2',
  linkColor: 'rgb(98, 174, 250)',
  gridBackgroundColor: '#181818',
  cellColor: '#222222',
  borderColor: '#4a4a4a',
  accentColor1: '#253543',
  accentColor2: '#345170',
  accentColor3: '#416f9c',
  accentColor4: '#4f8cc9',
  accentColor5: '#4f8cc9',
};

export const rainbow = {
  backgroundColor: `repeating-linear-gradient(45deg,
    rgba(255, 108, 108, 0.2) 0px, rgba(255, 108, 108, 0.2) 20px,
    rgba(255, 162, 42, 0.2) 20px, rgba(255, 162, 42, 0.2) 40px,
    rgba(123, 201, 111, 0.2) 40px, rgba(123, 201, 111, 0.2) 60px,
    rgba(94, 195, 220, 0.2) 60px, rgba(94, 195, 220, 0.2) 80px,
    rgba(193, 127, 214, 0.2) 80px, rgba(193, 127, 214, 0.2) 100px
  )`,
  textColor: '#24292e',
  secondaryTextColor: '#24292e',
  linkColor: '#615fd0',
  gridBackgroundColor: '#FFFFFF',
  cellColor: 'rgb(235, 237, 240)',
  borderColor: '#b2b7bb',
  accentColor1: 'rgba(255, 108, 108, 0.7)', // red
  accentColor2: 'rgba(255, 162, 42, 1)', // orange
  accentColor3: 'rgba(123, 201, 111, 1)', // green
  accentColor4: 'rgba(94, 195, 220, 1)', // blue
  accentColor5: 'rgba(165, 107, 183, 0.7)', // purple
};
