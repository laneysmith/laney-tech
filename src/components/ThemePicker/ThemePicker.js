import React from 'react';
import PropTypes from 'prop-types';
import { ThemePickerContainer, ThemeButton } from '../styles/ThemePickerStyles';

const ThemePicker = ({ themeList, theme, setTheme }) => (
  <ThemePickerContainer>
    <b>Theme:</b>
    {themeList.map((color) => {
      const isSelected = theme === color;
      return (
        <ThemeButton
          id={`${color}-theme-btn`}
          key={`${color}-theme`}
          type="button"
          onClick={() => setTheme(color)}
          disabled={isSelected}
        >
          {color}
        </ThemeButton>
      );
    })}
  </ThemePickerContainer>
);

ThemePicker.propTypes = {
  themeList: PropTypes.arrayOf(PropTypes.string).isRequired,
  theme: PropTypes.string.isRequired,
  setTheme: PropTypes.func.isRequired,
};

export default ThemePicker;
