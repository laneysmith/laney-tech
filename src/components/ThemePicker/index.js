export { default as ThemePicker } from './ThemePicker';
