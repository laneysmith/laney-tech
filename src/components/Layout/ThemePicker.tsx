import React from 'react';
import styled from 'styled-components';

import { THEME_LIST, Theme } from './themes';
import { rhythm } from '../../utils/typography';
import Button, { StyledButton } from '../Shared/Button';

const ThemePickerContainer = styled.div`
  text-align: right;
  width: 100%;

  @media only screen and (max-width: 600px) {
    padding-right: ${rhythm(0.5)};
  }

  ${StyledButton} {
    margin-left: ${rhythm(0.5)};
  }
`;

interface ThemePickersProps {
  theme: Theme;
  setTheme: (theme: Theme) => void;
}

const ThemePicker: React.FC<ThemePickersProps> = ({ theme, setTheme }) => {
  return (
    <ThemePickerContainer>
      <b>Theme:</b>
      {THEME_LIST.map(color => {
        const isSelected = theme === color;
        return (
          <Button key={color} onClick={() => setTheme(color)} aria-pressed={isSelected}>
            {color}
          </Button>
        );
      })}
    </ThemePickerContainer>
  );
};

export default ThemePicker;
