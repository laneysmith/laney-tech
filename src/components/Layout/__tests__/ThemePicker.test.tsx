import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import ThemePicker from '../ThemePicker';
import { THEME_LIST } from '../themes';

describe('ThemePicker', () => {
  it('should call setTheme with the correct values', () => {
    const setThemeMock = jest.fn();
    render(<ThemePicker theme={THEME_LIST[0]} setTheme={setThemeMock} />);

    userEvent.click(screen.getByLabelText('dark'));

    expect(setThemeMock).toHaveBeenCalledWith('dark');
    expect(setThemeMock.mock.calls.length).toBe(1);

    userEvent.click(screen.getByLabelText('rainbow'));

    expect(setThemeMock).toHaveBeenCalledWith('rainbow');
    expect(setThemeMock.mock.calls.length).toBe(2);
  });

  it('should have the correct theme selected by default', () => {
    const { rerender } = render(<ThemePicker theme={THEME_LIST[0]} setTheme={jest.fn()} />);
    const firstThemeRadio = screen.getByLabelText('light');
    const secondThemeRadio = screen.getByLabelText('dark');

    expect(firstThemeRadio).toBeChecked();

    rerender(<ThemePicker setTheme={jest.fn()} theme={THEME_LIST[1]} />);

    expect(secondThemeRadio).toBeChecked();
  });
});
