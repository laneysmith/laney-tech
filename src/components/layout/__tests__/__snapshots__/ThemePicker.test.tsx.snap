// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemePicker should match snapshot 1`] = `
.c1 {
  border: 1px solid;
  border-radius: 3px;
  font-size: 1em;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 6px 8px;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
}

.c1:disabled {
  opacity: 0.8;
}

.c1:disabled:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:focus {
  outline: none;
}

.c0 {
  text-align: right;
  width: 100%;
}

@media only screen and (max-width:600px) {
  .c0 {
    padding-right: 0.8125rem;
  }
}

<div
  class="c0"
>
  <b>
    Theme:
  </b>
  <button
    class="c1"
    disabled=""
    id="light-theme-btn"
    style="margin-left: 0.8125rem;"
  >
    light
  </button>
  <button
    class="c1"
    id="dark-theme-btn"
    style="margin-left: 0.8125rem;"
  >
    dark
  </button>
  <button
    class="c1"
    id="rainbow-theme-btn"
    style="margin-left: 0.8125rem;"
  >
    rainbow
  </button>
</div>
`;
